packages/connectors-web/src/alchemySmartWallet.ts

```diff
@@ -1,85 +1,233 @@
       type: alchemySmartWallet.type,
-      baseConnector,
+      baseConnector: innerConnector,
 
       async setup(): Promise<void> {
-        await baseConnector.setup?.();
-
-        // Optional function for running when the connector is first created.
-        throw new Error("Not implemented: setup");
+        await innerConnector.setup?.();
+        if (await innerConnector.isAuthorized()) {
+          await initializeSmartWalletClient();
+        }
       },
 
-      async connect(parameters) {
-        await baseConnector.connect(parameters);
-        throw new Error("Not implemented: connect");
+      async connect(params): Promise<{ accounts: Address[]; chainId: number }> {
+        resetState();
+        await innerConnector.connect(params);
+        await initializeSmartWalletClient();
+        const chainId = params?.chainId ?? (await innerConnector.getChainId());
+        const accounts = (await this.getAccounts()) as Address[];
+        config.emitter.emit("connect", { chainId, accounts });
+        return { accounts, chainId };
       },
 
-      async disconnect() {
-        await baseConnector.disconnect?.();
-        throw new Error("Not implemented: disconnect");
+      async disconnect(): Promise<void> {
+        await innerConnector.disconnect?.();
+        resetState();
+        config.emitter.emit("disconnect");
       },
 
-      async getAccounts() {
-        // TODO: Get the account from the signer
-        return [];
+      async getAccounts(): Promise<Address[]> {
+        const smartWalletClient = await getSmartWalletClientIfConnected();
+        const address = smartWalletClient?.account?.address;
+        return address ? [address] : [];
       },
 
-      async getChainId() {
-        // TODO: the connector will need to presist the chain id and instantiate the client/provider with it
-        throw new Error("Not implemented: getChainId");
+      async getChainId(): Promise<number> {
+        return innerConnector.getChainId();
       },
 
-      async getProvider() {
-        throw new Error("Not implemented: getProvider");
+      async getProvider(): Promise<Provider> {
+        const smartWalletClient = await getSmartWalletClientIfConnected();
+        return smartWalletClient?.getProvider() ?? loggedOutProvider;
       },
 
-      async isAuthorized() {
-        throw new Error("Not implemented: isAuthorized");
+      async isAuthorized(): Promise<boolean> {
+        return innerConnector.isAuthorized();
       },
 
-      async switchChain({ chainId }) {
-        void chainId;
-        throw new Error("Not implemented: switchChain");
+      async switchChain(params): Promise<Chain> {
+        const { chainId } = params;
+        const currentChainId = await this.getChainId();
+        if (chainId === currentChainId) {
+          return getChainFromConfig(config, chainId);
+        }
+        resetState();
+        const baseChain = await innerConnector.switchChain?.(params);
```

Comment:
Maybe not an issue but I wonder if it's worth reseting the state after successfully switching chains?

If the innerconnector switch chan is rejected then we will be stuck in a state where the `smartWalletClientPromise` is undefined.